### PR TITLE
Feature/bai 342 replace all references of request with approval

### PR DESCRIPTION
**Pull request by ARADDCC012**
Refactored the following:
Request -> Approval
RequestType -> ApprovalCategory

Some of the logging has been updated as a result. This is likely to mean we will need to update our translations HS.

Also fixed some typos I noticed while CTRL+F'ing the codebase. These were all in .md files so won't have any knock-on effects.

### CODE REVIEW

File: data/approvals.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import qs from 'qs'
+import useSWR from 'swr'
+import { Approval } from '../types/interfaces'
+import { fetcher } from '../utils/fetcher'
+
+export type ApprovalCategory = 'Upload' | 'Deployment'
+export type ReviewFilterType = 'user' | 'archived'
```
Comment (a3957273) [inline data/approvals.ts:7]: Should this be `ApprovalFilterType`?

File: server/migrations/002-user-array-support.ts [hunk 2]
```diff
@@ -52,27 +52,27 @@ export async function up() {
     await model.save()
   }
 
-  const requests = await RequestModel.find({})
+  const approvals = await ApprovalModel.find({})
```
Comment (a3957273) [inline server/migrations/002-user-array-support.ts:55]: This is a migration, so shouldn't be altered in this way.

**Review by a3957273 — CHANGES_REQUESTED** [review]

File: data/approvals.ts (hunk 1 above)
Comment (ARADDCC012) [reply data/approvals.ts:7]: I've not refactored "review"s, only "request"s. Do you also want to change all reviews to approvals?

File: server/migrations/002-user-array-support.ts (hunk 2 above)
Comment (ARADDCC012) [reply server/migrations/002-user-array-support.ts:55]: Yeah I wondered if we wanted to change it or not. It'll complain about `RequestModel` not existing anymore, but will revert any changes to this file.

**Review by a3957273 — APPROVED** [review]

**Pull request merged**